express/parsers/apps/espresso/parser.py

```diff
@@ -14,23 +14,38 @@
 from express.parsers.apps.espresso.formats.espresso_legacyxml import EspressoLegacyXMLParser
 
 
-class EspressoLegacyParser(BaseParser, IonicDataMixin, ElectronicDataMixin, ReciprocalDataMixin):
+class EspressoParser(BaseParser, IonicDataMixin, ElectronicDataMixin, ReciprocalDataMixin):
     """
     Espresso parser class.
     """
+    _xml_parsers = {
+        "data-file.xml": EspressoLegacyXMLParser,
+        "data-file-schema.xml": Espresso640XMLParser,
+    }
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.work_dir = self.kwargs["work_dir"]
         self.stdout_file = self.kwargs["stdout_file"]
         self.txt_parser = EspressoTXTParser(self.work_dir)
-        self.xml_data_file = settings.XML_DATA_FILE
+        self._xml_full_file_path = None
+        self._xml_data_file = None
         self._xml_parser = None
 
+    @property
+    def xml_data_file(self):
```

Comment:
Let's use the basename/dirname convention, pls

Reply:
I will update the XML file references for the espresso parser here, but generally updating to `basename` and `dirname` for consistency across the board will impact all parser implementations. If that is intended scope, I will track it as a separate issue.